File: 프로그래머스/3/43238. 입국심사/입국심사.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        
        long min = 0;
        long max = 0;
        
        for(int i=0; i<times.length; i++){
            max = Math.max(max, times[i]);
        }
        
        max *= n;
        
        long answer = 0;
        
        while(min <= max){
            
            long mid = (min + max)/2;
            long cnt = 0;
            
            for(int i=0; i<times.length; i++){
                cnt += mid/times[i];
            }
            
            if(cnt >= n) {
                answer = mid;
                max = mid -1;
            }else{
                min = mid + 1;
            }
            
        }
       
        return answer;
    }

}

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    times.sort((a,b) => b-a)
    var answer = 0
    var min = 0
    var max = Math.max(...times)*n
    
    while(min <= max){
        var mid = Math.floor((min+max)/2)
        const cnt = times.reduce((acc, time) => acc + Math.floor(mid/time), 0)
        if(cnt >= n){
            max = mid -1
            answer = mid
        }else{
            min = mid + 1
        }
    }
    
    
    return answer;
}